src/tracked_struct.rs

```diff
@@ -668,7 +679,6 @@ where
         &'db self,
         db: &'db dyn crate::Database,
         s: C::Struct<'db>,
-        _field_index: usize,
```

Comment:
You removing the `_field_index` here makes me wonder if we should update `revisions` to only store the revisions for tracked fields because they seem wasted for untracked fields. Removing the bookkeeping for untracked fields could improve memory usage because it reduces the metadata stored for every tracked struct instance. It may also reduce the need to pass both relative and absolute indices to `tracked_field`. 

However, this might also be something that's fine to follow up in a separate PR